static/nodejs_server/utils.ts

```diff
@@ -139,9 +148,13 @@ export function getChartUrl(
       if (!paramVal) {
         return;
       }
-      url += `${idx === 0 ? "" : "&"}${paramKey}=${paramVal}`;
+      let compressedVal = zlib.deflateSync(paramVal).toString("base64");
```

Comment:
How about having a helper for `compress` and `decompress` that wraps this COMPRESSED_VAL_ENCODING replacement?

Also, maybe have a simple unit-test that compresses, decompresses and confirms they match?

Reply:
done